### ISSUE TITLE
[Documentation/FAQ]

### ISSUE THREAD

**Issue by rubyFeedback** [Sharrisii/TAZ#6]
Hello Sharrisii,

If you have some spare time one day in the future, would it be possible to either add to the documentation here (README.md I suppose), or a new FAQ, what TAZ is about?

Questions could include:

- Can TAZ be installed persistently, such as a gentoo .iso too?

This is my only question for now, but of course some secondary aspects may be interesting, e. g. how TAZ came to be, if there are any speed advantages/disadvantages, recommendation ... and also how to use the 32 binary parts. (This may be interesting for people who use winehq)

Thanks for reading!

**Comment by Sharrisii** [comment]
No, it can't be installed persistently. This ability would somewhat counteract what TAZ is about, because:
* we want it to be usable together with your regular OS (i.e. windows or another linux distro, which is often installed to the harddisk -if you have one-). Also, it's easy to upgrade this without hassle; if on the harddisk, a software update is needed which can sometimes break the system or introduce flaws. Contrary to this, just inserting a newly written cd/usb stick is easy and hassle-free (and it's easier to revert back to the older version in case you don't like a newer version of TAZ in case there are new flaws in it)
* harddisks tend to be slower on older pc's then usb sticks/SD-cards, ... (because they generally have older harddisks, ie PATA harddisks); also they are less reliable due to the multiple moving parts, and with TAZ (especially the x86 version), we're focusing mainly on older pc's

**Comment by Sharrisii** [comment]
Here are some additional questions asked by users when I announced the TAZ release at the Pale Moon Forum:

Moonchild wrote:
Sun Feb 02, 2020 2:48 am
Not to diss the effort of making such a distro, but if you were going to include New Moon as default on it, you should at the very least try to use a current version and not cripple half the web with this insanity on top. :/ 

Yes, the version is still an older one. we never updated it since we implemented it the first time (remember that development has taken a looooong time, due to all those special things I added -the different boot setups took the most time btw-).
It would have taken a lot of time to update it, since the distro runs under musl (and palemoon doesn't compile easily with musl), hence the reason why we focused on the other things and just get a first release out as soon as possible.
One of our next tasks will be to allow it to compile under musl easily, in the manner explained by wolfbeast (there are still posts on this here; in particular reply: Sat Mar 10, 2018 3:04 pm ). After that, we can get a new palemoon version integrated in a next release.

Also remember there are still a lot of other unresolved issues (many of them quite important btw): https://github.com/Sharrisii/TAZ/issues/4
The hope is that the people that will switch to use this will also start to help us on correcting these issues (they're not that easy to solve). The fact that we are using github should help us on this matter as other users/developers can simply contribute to development of both the main iso and the optional usb stick (settings can be improved, plugins can be replaced easily by newer versions of those plugins, ...).

As for having both Ublock and noscript in TAZ (reason):
it's been a while since I compiled that plugin list (and again they may even be outdated a bit by now), but the idea I believe was that when you disable noscript (which you need to do to still allow some websites to actually run well), then ublock still helps out by blocking connectins/advertisements. In websites that don't require javascript to run, you can again enable noscript. So the plugins have indeed been chosen carefully, there is a system behind it.

New Tobin Paradigm wrote:
Sat Feb 01, 2020 10:50 pm
And what is this: https://github.com/Sharrisii/TAZ_option ... iption.txt , hmm? 

I wanted to add extra safety and special functions to palemoon. That said, do notice that this script is on the optional usb stick repo.
That means that when you write this to the usb stick (or compact flash for that matter as that would be faster), you can indeed modify the script to suit you (disabling whatever function you want or not, or even any plugin you want or not).
The usb stick is intented to allow the user to change settings, and store data (for those not having a harddisk), as any data read from the main iso (which is most often going to be written to a cd) is not changable, and the tmpfs (which this data is written to) can't store anything for a longer period either (data removed upon reboot).
Note btw that you can also run palemoon in (gentoo's own) sandbox but that is entirely optional and there are other things you can activate.

As for the obvious question that will no doubt follow: why would I not have a harddisk ?: these tend to be "more polluting" then usb sticks or compact flash (with that I mean that they're made of different materials and I doubt each material is taken apart to be recycled correctly, especially depending in what country you live). Also, especially with older pc's (that still use PATA, not SATA), they're also faster and tend to be more reliable (regular harddisks have moving parts).

loxodont wrote:
It's a good minimalist distro, usefull for USB drives/ traveling sticks and I'm sure it could run well on my very old 2-core Pentium and AMD 900 devices (as XP replacement), but even compared to my Puppies it is very minimalist for my taste. 

Taste differs for each person, but TAZ has been developed to not have any more programs then what you "really" need. I have no doubt that in standard config, Puppy Linux has far more things then you use on a daily basis. That's an important issue, because the more things you load (and not use it), the slower it makes your machine. On newer machines it may matter less, but especially on older machines, this could really make a difference. We also integrated conky so you can measure cpu stress for each boot setup; that way you can actually measure this speed gain between the boot setups, and even between distro's as on other distro's too you can install conky too.
This speed gain is also the reason also why there are multiple boot setups: you select whichever boot setup you need for that computing session and load it (they all have different program configurations), the most minimal one is the core boot setup (boot setup 1).
Want a different boot setup then what you loaded: then you need to restart. The fact that you need to restart/put in effort will also make you not stray from whatever you actually (initially) wanted to do that day (and not get distracted and run full length cat videos for instance).
For boot setup 2 to 6 to work, you need to manually copy over some files from the release to the usb stick (https://github.com/Sharrisii/TAZ_option ... tsetup2to6 ), have the usb stick inserted and recognized by machine, and make the correct choice of the boot setup you want at the boot menu. Also keep in mind that for still special programs you want to add there is the "7th boot setup". Either ask Stephanie to compile the custom programs you want into a boot setup (paying service) or do it yourself. We integrated this paying service to help fund development for TAZ (needed for extra releases and all of the other work, ie special musl development which again helps TAZ, ....).
